spec/lib/msf/core/modules/metadata/search_spec.rb

```diff
@@ -1,43 +1,78 @@
-RSpec.shared_examples_for 'Msf::Module::Search' do
-  describe '#search_filter' do
+require 'spec_helper'
+
+RSpec.describe Msf::Modules::Metadata::Search do
+  let(:mock_cache) do
+    context_described_class = described_class
+    Class.new do
+      include context_described_class
+
+      def get_metadata
+        # mocked
+      end
+    end
+  end
+
+  subject { mock_cache.new }
+
+  let(:default_opts) do
+    {
+      'name' => 'module',
+      'fullname' => '/module',
+      'mod_time' => '2020-08-14',
+      'references' => ['cve-1234-1111'],
+      'author' => %w(author1 author2)
+    }
+  end
+  let(:opts) { {} }
+  let(:mock_module) { Msf::Modules::Metadata::Obj.from_hash(default_opts.merge(opts)) }
+
+  before(:each) do
+    allow(subject).to receive(:get_metadata).and_return([mock_module])
+  end
+
+  describe '#find' do
     REF_TYPES = %w(CVE BID EDB)
 
     shared_examples "search_filter" do |opts|
       accept = opts[:accept] || []
       reject = opts[:reject] || []
 
+      def find(search_string)
+        search_params = described_class.parse_search_string(search_string)
+        subject.find(search_params)
+      end
+
+      # inverse the query string. An input such as `os:osx` will be converted to the inverse, i.e. everything but osx `os:-osx`
+      def inverse_query_terms(search_string)
+        search_string.gsub(':', ':-')
+      end
+
       accept.each do |query|
         it "should accept a query containing '#{query}'" do
-          # if the subject matches, search_filter returns false ("don't filter me out!")
-          expect(subject.search_filter(query)).to be_falsey
+          expect(find(query)).to eql([mock_module])
         end
 
         unless opts.has_key?(:test_inverse) and not opts[:test_inverse]
-          it "should reject a query containing '-#{query}'" do
-            expect(subject.search_filter("-#{query}")).to be_truthy
+          it "should reject an inverse query containing of '#{query}'" do
+            expect(find(inverse_query_terms(query))).to be_empty
```

Comment:
The new search cache has stricter validation, this test was previously producing invalid query strings such as:

```
-os:osx
```

Which isn't supported, currently the only functionality supported is `os:-osx`